controllers/kubevirtmachine_controller.go

```
@@ -354,6 +382,31 @@ func (r *KubevirtMachineReconciler) updateNodeProviderID(ctx *context.MachineCon
 }
 
 func (r *KubevirtMachineReconciler) reconcileDelete(ctx *context.MachineContext) (ctrl.Result, error) {
+	infraClusterClient, infraClusterNamespace, err := r.InfraCluster.GenerateInfraClusterClient(ctx.ClusterContext())
+	if err != nil {
+		ctx.Logger.Error(err, "Infra cluster client is not available.")
```

seems like we should return an error here.